scripts/irods/test/test_iput.py

```diff
@@ -513,6 +513,48 @@ def do_test_issue_5496_impl(self, file_size_in_bytes):
         self.assertEqual(len(err), 0)
         self.assertEqual(len(checksum.strip()), 0)
 
+    def test_iput_with_apostrophe_logical_path__issue_5759(self):
+        '''Tests iput with apostrophes in different places in the data object name
+
+        Each item is a combination of apostrophes at the beginning, in the middle, and at the end of the data object name.
+        Each of these is put to a collection with apostrophes at the beginning, in the middle, and at the end of the data object name.
+        '''
+
+        local_file = os.path.join(self.user.local_session_dir, 'test_iput_with_apostrophe_logical_path__issue_5759')
+        lib.make_file(local_file, 1024, 'arbitrary')
+
+        collection_names = ["collection",
+                            "collection'",
+                            "collect'ion",
+                            "collect'ion'",
+                            "'collection",
+                            "'collection'",
+                            "'collect'ion",
+                            "'collect'ion'"]
+
+        data_names = ["data_object",
+                      "data_object'",
+                      "data'_object",
+                      "data'_object'",
+                      "'data_object",
+                      "'data_object'",
+                      "'data'_object",
+                      "'data'_object'"]
+
+        for coll in collection_names:
+            collection_path = os.path.join(self.user.session_collection, coll)
+
+            self.user.assert_icommand(['imkdir', collection_path])
+
+            for name in data_names:
+                logical_path = os.path.join(collection_path, name)
+
+                self.user.assert_icommand(['iput', local_file, logical_path])
```

Comment:
does `ireg` need the same exercising?

Reply:
I started to write tests for several iCommands (`irepl`, `icp`, etc.) but backed them out since I was just testing the apostrophe bit (didn't seem important to me at the time to try all the different iCommands). I can add these back as well as one for `ireg`.

Reply:
I also noticed while contemplating this test that this does not cover all the permutations for `icp` in particular because there will be two data objects to consider, not just one: source and destination. So that test will need to cover an additional set of scenarios for the source data object.

Reply:
Done